### PR TITLE
refactor: add transformation_rule_id to _tool_jira_boards

### PR DESCRIPTION
**Pull request by mindlesscloud**
### Summary
In addition to fixing some omissions in #3753, this PR contains the following modification:
1. add `transformation_rule_id` to `_tool_jira_boards`
2. provide `PATCH` API for Jira scope

### Does this close any open issues?
No, but partially fulfilled issue #3468

### Screenshots
the scope API
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/jira/api/scope.go [hunk 1]
```diff
@@ -24,52 +24,66 @@ import (
...
+// @Accept application/json
+// @Param connectionId path int false "connection ID"
+// @Param boardId path int false "board ID"
+// @Param scope body models.JiraBoard true "json"
+// @Success 200  {object} models.JiraBoard
+// @Failure 400  {object} shared.ApiBody "Bad Request"
+// @Failure 500  {object} shared.ApiBody "Internal Error"
+// @Router /plugins/jira/connections/{connectionId}/scopes/{boardId} [PATCH]
+func UpdateScope(input *core.ApiResourceInput) (*core.ApiResourceOutput, errors.Error) {
+	board, err := extractBoard(input)
```
Comment (klesh) [inline plugins/jira/api/scope.go:69]: This is not needed. What we need to do:
1. load the board from db
2. DecodeMapStruct from input.body to the board (this is essentially a patch operation)
3. verify and save

File: plugins/jira/api/scope.go [hunk 2]
```diff
@@ -122,3 +149,16 @@ func extractParam(params map[string]string) (uint64, uint64) {
...
 }
+
+func verifyBoard(board *models.JiraBoard) errors.Error {
+	if board.ConnectionId == 0 {
+		return errors.BadInput.New("invalid connectionId")
+	}
+	if board.BoardId == 0 {
+		return errors.BadInput.New("invalid boardId")
+	}
+	if board.ScopeId != strconv.FormatUint(board.BoardId, 10) {
```
Comment (likyh) [inline plugins/jira/api/scope.go:160]: why not set `board.ScopeId = board.BoardId` instead reading it from frontend?

Comment (klesh) [reply plugins/jira/api/scope.go:160]: I think it is for config-ui to display the scope in a uniform way, especially on the pipeline page.
config-ui can display scope id/name directly without detecting plugin.

Comment (klesh) [reply plugins/jira/api/scope.go:160]: come to think about it, do we have `ScopeName` as well? config-ui would need this for a human-readable representation.

Comment (likyh) [reply plugins/jira/api/scope.go:160]: ScopeName is easy to set. But ScopeId maybe has some complex rules to calculate.

Comment (mindlesscloud) [reply plugins/jira/api/scope.go:160]: I wish to expose `scopeId`, `transformationRuleId`, and `connection_id` as common fields among all of the plugin's scope API

**Pull request merged**